### DISCUSSION TITLE
Consider publishing all source code action kinds so that it can be used in "editor.codeActionsOnSave"

### DISCUSSION THREAD

**Discussion by heejaechang** [microsoft/pylance-release#3073] [Enhancement]
vscode has this [```option```](https://code.visualstudio.com/updates/v1_23#_run-code-actions-on-save) where user can specify code action kinds to run on save.

code action kinds ```source.``` is supposed to work on whole source file so all source kind should work with the option. if not, at least publish all ```source.fixAll``` kind so that user can pick which one to run on save.

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.

**Comment by heejaechang** [comment]
related vscode issue - https://github.com/microsoft/vscode/issues/82718

**Comment by heejaechang** [comment]
related pylance issue - https://github.com/microsoft/pylance-release/issues/3091
